src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CARPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CARPACITY];
+
+    }
 
     @Override
     public void add(T value) {
-
+        sizeChecking();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        sizeChecking();
+        if (index == size) {
+            add(value);
+            return;
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        size++;
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexChecking(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexChecking(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexChecking(index);
+        T[] tempArray = (T[]) elementData;
+        final T removedValue = tempArray[index];
+        System.arraycopy(tempArray, 0, elementData, 0, index);
+        System.arraycopy(tempArray, index + 1, elementData, index, tempArray.length - 1 - index);
```

Comment:
you can make it with one System.arraycopy

Reply:
done